test/extended/networking/acl_audit_log.go

```diff
@@ -0,0 +1,213 @@
+			allowLogFound = true
+		}
+
+		if strings.Contains(logLine, "default-deny-all\", verdict=drop") && strings.Contains(logLine, "nw_src="+ip[2]) &&
+			strings.Contains(logLine, "nw_dst="+ip[0]) {
+			denyLogFound = true
+		}
+
+		if allowLogFound && denyLogFound {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("Acl logs exist but do not contain correct src dst ips and actions")
+}
+
+func launchACLLoggingPod(f *e2e.Framework, namespace string, nodeName string, podName string) error {
+	contName := fmt.Sprintf("%s-container", podName)
+	pod := &kapiv1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: namespace,
+		},
+		Spec: kapiv1.PodSpec{
+			Containers: []kapiv1.Container{
+				{
+					Name:  contName,
+					Image: "k8s.gcr.io/e2e-test-images/agnhost:2.21",
+					Args:  []string{"netexec", "--http-port=80", "--udp-port=90"},
+				},
+			},
+			NodeName:      nodeName,
+			RestartPolicy: kapiv1.RestartPolicyNever,
+		},
+	}
+	_, err := f.ClientSet.CoreV1().Pods(namespace).Create(context.Background(), pod, metav1.CreateOptions{})
+	return err
+}
+
+func waitForACLLoggingPod(f *e2e.Framework, namespace string, podName string) (string, error) {
+	podIP := ""
+	err := frameworkpod.WaitForPodCondition(f.ClientSet, namespace, podName, "running", podStartTimeout, func(pod *kapiv1.Pod) (bool, error) {
+		podIP = pod.Status.PodIP
+		return (podIP != "" && pod.Status.Phase != kapiv1.PodPending), nil
+	})
+	return podIP, err
+}
+
+func makeDenyAllPolicy(f *e2e.Framework) (*networkingv1.NetworkPolicy, error) {
+
+	policy := &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default-deny-all",
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{},
+			PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeEgress, networkingv1.PolicyTypeIngress},
+			Ingress:     []networkingv1.NetworkPolicyIngressRule{},
+			Egress:      []networkingv1.NetworkPolicyEgressRule{},
+		},
+	}
+
+	policy, err := f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(context.TODO(), policy, metav1.CreateOptions{})
+
+	return policy, err
+}
+
+func makeFromSameNSPolicy(f *e2e.Framework) (*networkingv1.NetworkPolicy, error) {
+
+	policy := &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "allow-from-same-ns",
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{},
+			PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{{
+				From: []networkingv1.NetworkPolicyPeer{{
+					PodSelector: &metav1.LabelSelector{},
+				}},
+			}},
+		},
+	}
+
+	policy, err := f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(context.TODO(), policy, metav1.CreateOptions{})
+
+	return policy, err
+}
+
+func cleanupNetworkPolicy(f *e2e.Framework, policy *networkingv1.NetworkPolicy) {
```

Comment:
.do you need to do this? they'll go away when the Namespace is deleted

Reply:
Nope removed :)